diff --git a/node_modules/@types/node/ts4.8/buffer.d.ts b/node_modules/@types/node/ts4.8/buffer.d.ts
index ea859cd..30eca87 100755
--- a/node_modules/@types/node/ts4.8/buffer.d.ts
+++ b/node_modules/@types/node/ts4.8/buffer.d.ts
@@ -2206,7 +2206,6 @@ declare module 'buffer' {
              */
             values(): IterableIterator<number>;
         }
-        var Buffer: BufferConstructor;
         /**
          * Decodes a string of Base64-encoded data into bytes, and encodes those bytes
          * into a string using Latin-1 (ISO-8859-1).
@@ -2241,17 +2240,6 @@ declare module 'buffer' {
         function btoa(data: string): string;
 
         interface Blob extends __Blob {}
-        /**
-         * `Blob` class is a global reference for `require('node:buffer').Blob`
-         * https://nodejs.org/api/buffer.html#class-blob
-         * @since v18.0.0
-         */
-        var Blob: typeof globalThis extends {
-            onmessage: any;
-            Blob: infer T;
-        }
-            ? T
-            : typeof NodeBlob;
     }
 }
 declare module 'node:buffer' {
diff --git a/node_modules/@types/node/ts4.8/dom-events.d.ts b/node_modules/@types/node/ts4.8/dom-events.d.ts
index b9c1c3a..64f1be5 100755
--- a/node_modules/@types/node/ts4.8/dom-events.d.ts
+++ b/node_modules/@types/node/ts4.8/dom-events.d.ts
@@ -105,22 +105,10 @@ import {} from 'events';    // Make this an ambient declaration
 declare global {
     /** An event which takes place in the DOM. */
     interface Event extends __Event {}
-    var Event: typeof globalThis extends { onmessage: any, Event: infer T }
-        ? T
-        : {
-            prototype: __Event;
-            new (type: string, eventInitDict?: EventInit): __Event;
-        };
 
     /**
      * EventTarget is a DOM interface implemented by objects that can
      * receive events and may have listeners for them.
      */
     interface EventTarget extends __EventTarget {}
-    var EventTarget: typeof globalThis extends { onmessage: any, EventTarget: infer T }
-        ? T
-        : {
-            prototype: __EventTarget;
-            new (): __EventTarget;
-        };
 }
diff --git a/node_modules/@types/node/ts4.8/globals.d.ts b/node_modules/@types/node/ts4.8/globals.d.ts
index f401d95..fd8f641 100755
--- a/node_modules/@types/node/ts4.8/globals.d.ts
+++ b/node_modules/@types/node/ts4.8/globals.d.ts
@@ -24,23 +24,6 @@ interface NodeRequire extends NodeJS.Require { }
 interface RequireResolve extends NodeJS.RequireResolve { }
 interface NodeModule extends NodeJS.Module { }
 
-declare var process: NodeJS.Process;
-declare var console: Console;
-
-declare var __filename: string;
-declare var __dirname: string;
-
-declare var require: NodeRequire;
-declare var module: NodeModule;
-
-// Same as module.exports
-declare var exports: any;
-
-/**
- * Only available if `--expose-gc` is passed to the process.
- */
-declare var gc: undefined | (() => void);
-
 //#region borrowed
 // from https://github.com/microsoft/TypeScript/blob/38da7c600c83e7b31193a62495239a0fe478cb67/lib/lib.webworker.d.ts#L633 until moved to separate lib
 /** A controller object that allows you to abort one or more DOM requests as and when desired. */
@@ -64,16 +47,6 @@ interface AbortSignal extends EventTarget {
     readonly aborted: boolean;
 }
 
-declare var AbortController: {
-    prototype: AbortController;
-    new(): AbortController;
-};
-
-declare var AbortSignal: {
-    prototype: AbortSignal;
-    new(): AbortSignal;
-    // TODO: Add abort() static
-};
 //#endregion borrowed
 
 //#region ArrayLike.at()
diff --git a/node_modules/@types/node/ts4.8/url.d.ts b/node_modules/@types/node/ts4.8/url.d.ts
index e172acb..2ffb6e7 100755
--- a/node_modules/@types/node/ts4.8/url.d.ts
+++ b/node_modules/@types/node/ts4.8/url.d.ts
@@ -868,28 +868,6 @@ declare module 'url' {
             URL: typeof _URL;
             URLSearchParams: typeof _URLSearchParams;
         }
-        /**
-         * `URL` class is a global reference for `require('url').URL`
-         * https://nodejs.org/api/url.html#the-whatwg-url-api
-         * @since v10.0.0
-         */
-        var URL: typeof globalThis extends {
-            onmessage: any;
-            URL: infer T;
-        }
-            ? T
-            : typeof _URL;
-        /**
-         * `URLSearchParams` class is a global reference for `require('url').URLSearchParams`
-         * https://nodejs.org/api/url.html#class-urlsearchparams
-         * @since v10.0.0
-         */
-        var URLSearchParams: typeof globalThis extends {
-            onmessage: any;
-            URLSearchParams: infer T;
-        }
-            ? T
-            : typeof _URLSearchParams;
     }
 }
 declare module 'node:url' {
diff --git a/node_modules/@types/node/ts4.8/util.d.ts b/node_modules/@types/node/ts4.8/util.d.ts
index 6976c2b..46befd2 100755
--- a/node_modules/@types/node/ts4.8/util.d.ts
+++ b/node_modules/@types/node/ts4.8/util.d.ts
@@ -1130,31 +1130,6 @@ declare module 'util' {
     }
 
     import { TextDecoder as _TextDecoder, TextEncoder as _TextEncoder } from 'util';
-    global {
-        /**
-         * `TextDecoder` class is a global reference for `require('util').TextDecoder`
-         * https://nodejs.org/api/globals.html#textdecoder
-         * @since v11.0.0
-         */
-         var TextDecoder: typeof globalThis extends {
-            onmessage: any;
-            TextDecoder: infer TextDecoder;
-        }
-            ? TextDecoder
-            : typeof _TextDecoder;
-
-        /**
-         * `TextEncoder` class is a global reference for `require('util').TextEncoder`
-         * https://nodejs.org/api/globals.html#textencoder
-         * @since v11.0.0
-         */
-         var TextEncoder: typeof globalThis extends {
-            onmessage: any;
-            TextEncoder: infer TextEncoder;
-        }
-            ? TextEncoder
-            : typeof _TextEncoder;
-    }
 
     //// parseArgs
     /**
diff --git a/node_modules/@types/node/ts4.8/worker_threads.d.ts b/node_modules/@types/node/ts4.8/worker_threads.d.ts
index 52f4384..e93b046 100755
--- a/node_modules/@types/node/ts4.8/worker_threads.d.ts
+++ b/node_modules/@types/node/ts4.8/worker_threads.d.ts
@@ -646,43 +646,6 @@ declare module 'worker_threads' {
         MessageChannel as _MessageChannel,
         MessagePort as _MessagePort,
     } from 'worker_threads';
-    global {
-        /**
-         * `BroadcastChannel` class is a global reference for `require('worker_threads').BroadcastChannel`
-         * https://nodejs.org/api/globals.html#broadcastchannel
-         * @since v18.0.0
-         */
-        var BroadcastChannel: typeof globalThis extends {
-            onmessage: any;
-            BroadcastChannel: infer T;
-        }
-            ? T
-            : typeof _BroadcastChannel;
-
-        /**
-         * `MessageChannel` class is a global reference for `require('worker_threads').MessageChannel`
-         * https://nodejs.org/api/globals.html#messagechannel
-         * @since v15.0.0
-         */
-        var MessageChannel: typeof globalThis extends {
-            onmessage: any;
-            MessageChannel: infer T;
-        }
-            ? T
-            : typeof _MessageChannel;
-
-        /**
-         * `MessagePort` class is a global reference for `require('worker_threads').MessagePort`
-         * https://nodejs.org/api/globals.html#messageport
-         * @since v15.0.0
-         */
-         var MessagePort: typeof globalThis extends {
-            onmessage: any;
-            MessagePort: infer T;
-        }
-            ? T
-            : typeof _MessagePort;
-    }
 }
 declare module 'node:worker_threads' {
     export * from 'worker_threads';
